src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -58,15 +64,36 @@
     @Override
     public Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTags) {
         // Don't allow producing messages if we're beyond the storage limit.
-        long currentStorageUsage = storageUsed.get();
-        if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage > storageQuotaSoft && currentStorageUsage < storageQuotaHard) {
-            double minThrottle = quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
-            double limit = minThrottle * (1.0 - (1.0 * (currentStorageUsage - storageQuotaSoft) / (storageQuotaHard - storageQuotaSoft)));
-            maybeLog(lastLoggedMessageSoftTimeMs, "Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
-            return limit;
-        } else if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage >= storageQuotaHard) {
-            maybeLog(lastLoggedMessageHardTimeMs, "Limiting producer rate because disk is full. Used: {}. Limit: {}", storageUsed, storageQuotaHard);
-            return 1.0;
+        if (ClientQuotaType.PRODUCE.equals(quotaType)) {
+            Map<FileStore, Double> volumePercents = storagePercents.get();
+            for (Map.Entry<FileStore, Double> entry : volumePercents.entrySet()) {
+                Map.Entry<Double, Double> percents = percentsMap.get(entry.getKey());
+                if (percents != null) {
+                    Double currentPercent = entry.getValue();
+                    Double softPercent = percents.getKey();
+                    Double hardPercent = percents.getValue();
+                    if (currentPercent > softPercent && currentPercent < hardPercent) {
```

Comment:
Reorder the if/else: `if (currentPercent >= hardPercent) ... else if (currentPercent > softPercent) ...` saves a comparison.